### ISSUE TITLE
FES misses error cases

### ISSUE THREAD

**Issue by Spongman** [processing/p5.js#2740]
p5 0.6.0

these are some more errors that the current FES system fails to catch.

FES doesn't currently issue _any_ warning about any of these errors:
```javascript
color(0,0,0,'A'); // wrong type for optional parameter
color([0,0,0],0); // superfluous parameter 
color(['A','B','C']); // wrong element types
rect(0,0,0,0,null,0,0,0); // null, non-trailing, optional parameter
```

and it warns about a type error in this case:
```javascript
color('A','A',0,0,0,0,0,0,0,0);
```
where an argument count error would probably be more appropriate.

**Comment by himanshuc3** [comment]
Hi. I want to start contributing to p5js. Would you be kind enough to tell me things i need to learn. 
P.S. I know javascript.

**Comment by limzykenneth** [comment]
Hi @himanshuc3, your question is a bit unrelated to this issue but do see [this page](https://github.com/processing/p5.js/wiki/Development) to get started contributing to p5.js, you can also checkout issues marked with "good first issue" to start with. If you have further question do open another issue. Thanks!

**Comment by ankur54** [comment]
Hi. Is this issue still open? I would like to contribute to this issue.

**Comment by Spongman** [comment]
I think this was fixed a while back in the FES ‘scoring’ PR.

**Comment by lmccart** [comment]
thanks @Spongman. I've tested and confirmed this is fixed in 0.8.0, all four cases mentioned are now caught.
